libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpMPEG4Reader.java

```diff
@@ -0,0 +1,139 @@
+  private long firstReceivedTimestamp;
+  private int previousSequenceNumber;
+  private long startTimeOffsetUs;
+  private int sampleLength;
+
+  /** Creates an instance. */
+  public RtpMPEG4Reader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    previousSequenceNumber = C.INDEX_UNSET;
+    sampleLength = 0;
+  }
+
+  private static long toSampleUs(
+      long startTimeOffsetUs, long rtpTimestamp, long firstReceivedRtpTimestamp) {
+    return startTimeOffsetUs
+        + Util.scaleLargeTimestamp(
+            (rtpTimestamp - firstReceivedRtpTimestamp),
+            /* multiplier= */ C.MICROS_PER_SECOND,
+            /* divisor= */ MEDIA_CLOCK_FREQUENCY);
+  }
+
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_VIDEO);
+    castNonNull(trackOutput).format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    Log.i(TAG, "RtpMPEG4Reader onReceivingFirstPacket");
+  }
+
+  @Override
+  public void consume(ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker)
+      throws ParserException {
+    if (previousSequenceNumber != C.INDEX_UNSET && sequenceNumber != (previousSequenceNumber + 1)) {
+      Log.e(TAG, "Packet loss");
+    }
+    checkStateNotNull(trackOutput);
+
+    int limit = data.bytesLeft();
+    trackOutput.sampleData(data, limit);
+    sampleLength += limit;
+    parseVopType(data);
+
+    // Marker (M) bit: The marker bit is set to 1 to indicate the last RTP
+    // packet(or only RTP packet) of a VOP. When multiple VOPs are carried
+    // in the same RTP packet, the marker bit is set to 1.
+    if (rtpMarker) {
+        if (firstReceivedTimestamp == C.TIME_UNSET) {
+          firstReceivedTimestamp = timestamp;
+        }
+
+        long timeUs = toSampleUs(startTimeOffsetUs, timestamp, firstReceivedTimestamp);
+        trackOutput.sampleMetadata(timeUs, bufferFlags, sampleLength, 0, null);
+        sampleLength = 0;
+      }
+
+    previousSequenceNumber = sequenceNumber;
+  }
+
+  /**
+   * Parses VOP Coding type
+   *
+   * Sets {@link #bufferFlags} according to the VOP Coding type.
+   */
+  private void parseVopType(ParsableByteArray data) {
+    // search for VOP_START_CODE (00 00 01 B6)
+    byte[] inputData = data.getData();
+    byte[] startCode = {0x0, 0x0, 0x01, (byte) 0xB6};
+    int vopStartCodePos = Bytes.indexOf(inputData, startCode);
+    if (vopStartCodePos != -1) {
+      data.setPosition(vopStartCodePos + 4);
+      int vopType = data.peekUnsignedByte() >> 6;
+      bufferFlags = getBufferFlagsFromVopType(vopType);
+    }
+  }
+
+  @C.BufferFlags
+  private static int getBufferFlagsFromVopType(int vopType) {
+    return vopType == I_VOP ? C.BUFFER_FLAG_KEY_FRAME : 0;
+  }
+
+  @Override
+  public void seek(long nextRtpTimestamp, long timeUs) {
```

Comment:
Move this method so all `@Override` methods are grouped, in the same order as they are defined in `RtpPayloadReader`.